### ISSUE TITLE
[UX] Individual unit's kills statistic

### ISSUE THREAD

**Issue by 2018boyan** [wesnoth/wesnoth#6720]
### Describe the desired feature

I read from forums (https://forums.wesnoth.org/viewtopic.php?t=55571) and really think it is a great idea. The list will contain every unit () in a battle, and show how many enemies they kill. A further improvement is that when clicking a unit, it shows details, for example, the victim’s name, level, and even his kill list. Thanks for your work, if it's possible, this survey is helpful and funny for reviewing the battle.

**Comment by knyghtmare** [comment]
For UMC's I think it's quite doable

![image](https://user-images.githubusercontent.com/5283677/169667809-fbe6680a-82fd-4bc8-8c1e-7f69e48ccb74.png)

You may need some Lua and UI coding knowledge to start things off.

**Comment by Descacharrado** [comment]
I agree, It is not a big deal to track kills per campaign, but adding it as a feature to the game may be more tricky.

**Comment by knyghtmare** [comment]
> I agree, It is not a big deal to track kills per campaign, but adding it as a feature to the game may be more tricky.

I think what users want is for this to be made as an addon instead of the core game feature. Making an addon to show this is not difficult. Here is the basic outline.
1. add a var `kill_counter` to unit on creation (unit placed/recruit/recall)
2. if any kill, increment counter by 1
3. add UI to show the variable values for each unit (or use [message][option])
4. add some way to access it (right-click menu) item

 > A further improvement is that when clicking a unit, it shows details, for example, the victim’s name, level, and even himself's kill list. 

This is...requires a bit more intricacy (including the consent to enlarge save file size)..So, I recommend to make a working base first and then maybe pursue it.

**Comment by 2018boyan** [comment]
I get it, thank you so much! As a programming beginner, I should learn Lua and UI coding knowledge first.

**Comment by knyghtmare** [comment]
> I get it, thank you so much! As a programming beginner, I should learn Lua and UI coding knowledge first.

I think you may be overthinking the problem. The UI code already exists. Either make [message][option] to show the kill list or use the [choice_box] of the WLP. I think only WML knowledge is enough for this.
